hazelcast/src/main/java/com/hazelcast/cardinality/impl/CardinalityEstimatorContainer.java

```diff
@@ -69,23 +69,23 @@ public int getTotalBackupCount() {
     }
 
     /**
-     * Merges the given {@link SplitBrainMergeEntryView} via the given {@link SplitBrainMergePolicy}.
+     * Merges the given {@link KeyMergeDataHolder} via the given {@link SplitBrainMergePolicy}.
      *
-     * @param mergingEntry the {@link SplitBrainMergeEntryView} instance to merge
-     * @param mergePolicy  the {@link SplitBrainMergePolicy} instance to apply
+     * @param mergeData   the {@link KeyMergeDataHolder} instance to merge
+     * @param mergePolicy the {@link SplitBrainMergePolicy} instance to apply
      * @return the used {@link HyperLogLog} if merge is applied, otherwise {@code null}
      */
-    public HyperLogLog merge(SplitBrainMergeEntryView<String, HyperLogLog> mergingEntry, SplitBrainMergePolicy mergePolicy) {
-        String name = mergingEntry.getKey();
+    public HyperLogLog merge(KeyMergeDataHolder<String, HyperLogLog> mergeData, SplitBrainMergePolicy mergePolicy) {
+        String name = mergeData.getKey();
```

Comment:
Do you need to call `mergeData.setSerializationService()` and `existingData.setSerializationService()` here? In case someone implements a merge policy that calls `getDeserializedKey` or `getDeserializedValue`

Reply:
Actually at the moment you cannot have a custom merge policy for the `CardinalityEstimator` (see `CardinalityEstimatorConfig` for the very hard check on the allowed merge policies).

But if we change this in the future, we could easily forget about these two lines. So right now they are not needed, but I would still keep them (since they don't hurt much).